assets/js/pokeapi.js

```diff
@@ -0,0 +1,32 @@
+const pokeApi = {}
+
+function convertPokeApiDetailToPokemon(pokeDetail) {
+    const pokemon = new Pokemon()
+    pokemon.number = pokeDetail.order
+    pokemon.name = pokeDetail.name
+
+    const types = pokeDetail.types.map(typeSlot => typeSlot.type.name)
+    const [type] = types
+    pokemon.types = types
+    pokemon.type = type
+
+    pokemon.image = pokeDetail.sprites.other.dream_world.front_default
+
+    return pokemon
+}
+
+pokeApi.getPokemonDetail = (pokemon) => {
+    return fetch(pokemon.url)
+            .then(response => response.json())
+            .then(convertPokeApiDetailToPokemon)
+}
+
+pokeApi.getPokemons = (offset = 0, limit = 10) => {
+    const url = `https://pokeapi.co/api/v2/pokemon?offset=${offset}&limit=${limit}`;
+    return fetch(url)

```

Comment:
Maybe we could encapsulate the fetch function into a generic async fetch function so you could use for the other places as well using the async await syntax? Not required but good to have